library/std/src/error.rs

```diff
@@ -807,3 +807,206 @@ impl dyn Error + Send + Sync {
         })
     }
 }
+
+/// An error reporter that exposes the entire error chain for printing.
+/// It also exposes options for formatting the error chain, either entirely on a single line,
+/// or in multi-line format with each cause in the error chain on a new line.
+/// `Report` only requires that the wrapped error implements `Error`. It doesn't require that the
+/// wrapped error be `Send`, `Sync`, or `'static`.
+///
+/// # Examples
+///
+/// ```rust
+/// #![feature(error_reporter)]
+/// #![feature(negative_impls)]
+///
+/// use std::error::{Error, Report};
+/// use std::fmt;
+///
+/// #[derive(Debug)]
+/// struct SuperError<'a> {
+///     side: &'a str,
```

Comment:
See https://github.com/rust-lang/rust/pull/90174#discussion_r767278303

> What does `side` mean here? Is that left over from another example with `SideKick`?

Reply:
I ended up completely refactoring this example because the docs mentioned formatting multiple lines but it was using a string as a source (that's what side represents here I believe) instead of an actual source error, so I just copied the main example from the Error trait and used that to demonstrate the report output. In the future I want to completely rework the examples in this module to be closer to realistic usage of the error trait, but for now I'm going to use the existing example and focus on consistency.